subsys/bluetooth/host/l2cap.c

```diff
@@ -427,10 +427,27 @@ static struct net_buf *l2cap_create_le_sig_pdu(struct net_buf *buf,
 	return buf;
 }
 
+/* Send the buffer and release it in case of failure.
+ * Any other cleanup in failure to send should be handled by the disconnected
+ * handler.
+ */
+static inline void l2cap_send(struct bt_conn *conn, uint16_t cid,
+			      struct net_buf *buf)
+{
+	if (bt_l2cap_send(conn, cid, buf)) {
+		net_buf_unref(buf);
+	}
+}
+
 #if defined(CONFIG_BT_L2CAP_DYNAMIC_CHANNEL)
 static void l2cap_chan_send_req(struct bt_l2cap_chan *chan,
 				struct net_buf *buf, k_timeout_t timeout)
 {
+	if (bt_l2cap_send(chan->conn, BT_L2CAP_CID_LE_SIG, buf)) {
+		net_buf_unref(buf);
+		return;
```

Comment:
I'm a bit uncertain if this is the best thing to do, or if we should submit the RTX timer anyway.